web/test/import.test.ts

```diff
@@ -0,0 +1,210 @@
+
+  importProjectData = jest.fn()
+  simpleCreateProjectMeta = jest.fn()
+
+  onStep = jest.fn()
+
+  baseRootState = mockBaseRootState
+
+  mockMigrationData = JSON.stringify(sampleGoodDataExport)
+
+  mockGetState = jest.fn().mockReturnValue(baseRootState)
+
+  store = {
+    dispatch: jest.fn(),
+    getState: mockGetState,
+  }
+})
+
+describe('importProjectsData()', () => {
+  it('correctly calls functions with expected arguments', async () => {
+    await internalImportProjectsData(
+      getAppWs,
+      createProfilesZomeApi,
+      installProjectAppAndImport,
+      installProjectApp,
+      store,
+      mockMigrationData,
+      onStep
+    )
+
+    expect(getAppWs).toHaveBeenCalledTimes(1)
+
+    expect(createProfilesZomeApi).toHaveBeenCalledTimes(1)
+    expect(createProfilesZomeApi).toHaveBeenCalledWith({})
+
+    // make sure the test data is actually set up the way the test
+    // expects it to be
+    expect(sampleGoodDataExport.projects.length).toBe(2)
+    expect(sampleGoodDataExport.projects[0].projectMeta.isMigrated).toBeNull()
+    expect(
+      sampleGoodDataExport.projects[1].projectMeta.isMigrated
+    ).not.toBeNull()
+
+    expect(installProjectAppAndImport).toHaveBeenCalledTimes(1)
+    expect(installProjectAppAndImport).toHaveBeenCalledWith(
+      'testAgentAddress',
+      sampleGoodDataExport.projects[0],
+      sampleGoodDataExport.projects[0].projectMeta.passphrase,
+      store.dispatch
+    )
+
+    expect(installProjectApp).toHaveBeenCalledTimes(1)
+    expect(installProjectApp).toHaveBeenCalledWith(
+      sampleGoodDataExport.projects[1].projectMeta.passphrase
+    )
+
+    expect(store.dispatch).toHaveBeenCalled()
+    expect(store.getState).toHaveBeenCalledTimes(1)
+
+    expect(onStep).toHaveBeenCalledTimes(
+      1 + sampleGoodDataExport.projects.length
+    )
+  })
+
+  it('throws error with invalid json', async () => {
+    mockMigrationData = 'invalid json'
+    try {
+      await internalImportProjectsData(
+        getAppWs,
+        createProfilesZomeApi,
+        installProjectAppAndImport,
+        installProjectApp,
+        store,
+        mockMigrationData,
+        onStep
+      )
+    } catch (e) {
+      expect(e).toBeInstanceOf(SyntaxError)
+      expect(e.message).toBe('Unexpected token i in JSON at position 0')
+    }
+
+    mockMigrationData = null
+    try {
+      await internalImportProjectsData(
+        getAppWs,
+        createProfilesZomeApi,
+        installProjectAppAndImport,
+        installProjectApp,
+        store,
+        mockMigrationData,
+        onStep
+      )
+    } catch (e) {
+      expect(e).toBeInstanceOf(TypeError)
+      expect(e.message).toBe(
+        "Cannot read properties of null (reading 'projects')"
+      )
+    }
+  })
+})
+
+describe('installProjectAppAndImport()', () => {
+  const agentAddress = 'testAgentAddress'
+  it('correctly calls functions with expected arguments', async () => {
```

Comment:
Describe the intent of the function and its behavior in this scenario more specifically